content/blog/open-source-monitoring-at-the-hpe-customer-technology-center-böblingen.md

```diff
@@ -0,0 +1,377 @@
+
+```yaml
+apiVersion: v1 
+kind: ConfigMap 
+metadata: 
+  name: iloprom1-xml 
+  namespace: svtprometheus 
+data: 
+  iloprometheus.key: |- 
+    LVkjkV5j40Fl6U0O7ratU0RDf5W1NmAE0oqoQ2DaE74=
+  iloprometheus.xml: |- 
+    <data> 
+      <username>Prometheus</username>
+      <user>gAAAAABig6ZJgST83R7v5AFjsOKiYt0bdledJpF3-TnUBR5871eNttG8P-O_eYA55-5NyyZgmmeWRnWy3hPAbUFk18_sWNw==</user>
+      <password>gAAAAABig6ZJAZL8wS41b0snZ76VWo17JysUq0-k5NUCU1y5AAPw5p1MphkFcB_u6z0TgcPbmR9yJcX39Cx0Egwv5Lou8A==</password>
+      <monitoringintervall>30</monitoringintervall>
+      <logfile>iloprometheus.log</logfile>
+      <port>9091</port>
+      <server> <ILO_ip>10.1.40.5</ILO_ip>,<Rack>i5</Rack>,<Loc>16</Loc> </server>
+      <server> <ILO_ip>10.1.40.6</ILO_ip>,<Rack>i5</Rack>,<Loc>14</Loc> </server>  
+      <server> <ILO_ip>10.1.40.7</ILO_ip>,<Rack>i5</Rack>,<Loc>19</Loc> </server> 
+      <server> <ILO_ip>10.1.40.8</ILO_ip>,<Rack>i5</Rack>,<Loc>18</Loc> </server> 
+      <server> <ILO_ip>10.0.44.11</ILO_ip>,<Rack>F12</Rack>,<Loc>24</Loc> </server> 
+      <server> <ILO_ip>10.0.44.12</ILO_ip>,<Rack>F12</Rack>,<Loc>22</Loc> </server> 
+      <server> <ILO_ip>10.0.44.13</ILO_ip>,<Rack>F12</Rack>,<Loc>20</Loc> </server>     
+      <server> <ILO_ip>10.1.24.66</ILO_ip>,<Rack>C5</Rack>,<Loc>30</Loc> </server>     
+      <server> <ILO_ip>10.0.43.14</ILO_ip>,<Rack>i12</Rack>,<Loc>9</Loc> </server>
+    </data>
+```
+
+As you can see the config map contains two data parts: the iloprometheus.key and the iloprometheus.xml section. The first one is storing the encryption key, while the second one is storing user and password encrypted with the encryption key iloprometheus.key, the monitoring intervall, the logfile name and the port used by the exporter together with the server data. I used the rack location in addition to the ILO IP address in order to be able to visualize a rack temperature diagram if needed.
+
+T﻿he config map was stored in a YAML-file (iloprom1.yml) and then applied in my Kubernetes environment:
+
+```shell
+kubectl apply -f iloprom1.yml
+```
+
+With the config map in place, I could define the Kubernetes pod for the HPE Server exporter. The pod is using the iloprometheus:v4.3.1 container image, mounting the config map data to the /opt/prometheus/data directory in the container and is starting the Python script /opt/prometheus/iloPromConnector.v4.3.py.
```

Comment:
Consider rephrasing the first sentence to "With the config map in place, I was able to define the Kubernetes pod..."
Change the second sentence to "The pod uses the iloprometheus:...." and instead of "is starting the Python" change to "starts the Python"